tests/bearlib/abstractions/external_bear_wrap_testfiles/ExternalBearWrapTest.py

```diff
@@ -0,0 +1,105 @@
+import os
+import json
+import unittest
+
+from coalib.bearlib.abstractions.ExternalBearWrap import external_bear_wrap
+from coalib.results.Diff import Diff
+from coalib.results.Result import Result
+from coalib.settings.Section import Section
+
+
+def get_testfile_name(name):
+    """
+    Gets the full path to a testfile inside the same directory.
+
+    :param name: The filename of the testfile to get the full path for.
+    :return:     The full path to given testfile name.
+    """
+    return os.path.join(os.path.dirname(os.path.realpath(__file__)),
+                        name)
+
+
+class ExternalBearWrapComponentTest(unittest.TestCase):
+
+    class Dummy:
+        pass
+
+    class TestBear:
+
+        @staticmethod
+        def get_classfile_path():
+            return __file__
+
+    def setUp(self):
+        self.section = Section("TEST_SECTION")
+
+        self.test_program_path = get_testfile_name("test_external_bear.py")
+
+        self.testfile_path = get_testfile_name("test_file.txt")
+        with open(self.testfile_path, mode="r") as fl:
+            self.testfile_content = fl.read().splitlines(keepends=True)
+
+    def test_decorator_invalid_parameters(self):
+        with self.assertRaises(ValueError) as cm:
+            external_bear_wrap("exec", invalid_arg=88)
+        self.assertEqual(
+            str(cm.exception),
+            "Invalid keyword arguments provided: 'invalid_arg'")
+
+    def test_decorator_invalid_parameter_types(self):
+        # Provide some invalid severity maps.
+        with self.assertRaises(TypeError):
+            external_bear_wrap(executable=1337)
+
+    def test_get_executable(self):
+        uut = external_bear_wrap("exec")(self.TestBear)
+        self.assertEqual(os.path.basename(uut.get_executable()), "exec")
+
+    def test_get_executable_fail(self):
+        uut = external_bear_wrap("exec")(self.Dummy)
+        with self.assertRaises(NotImplementedError):
+            self.assertEqual(os.path.basename(uut.get_executable()), "exec")
+
+    def test_invalid_output(self):
+        broken_json = json.dumps([{'broken': "JSON"}])[:-1]
+        uut = (external_bear_wrap("exec")(self.Dummy)(self.section, None))
+        with self.assertRaises(ValueError):
+            # Something needs to be done with the result otherwise
+            # parse_output will not yield and thus will not raise the ValueError
+            list(uut.parse_output(broken_json, "some_file"))
+
+    def test_setting_desc(self):
```

Comment:
What happens when a non optional param is not available? Could you add a test for that.